test cases/common/163 disabler/meson.build

```diff
@@ -76,3 +76,25 @@ else
     has_not_changed = true
 endif
 assert(has_not_changed, 'App has changed.')
+
+if_is_disabled = true
+if disabler()
+  if_is_disabled = false
+else
+  if_is_disabled = false
+endif
+assert(if_is_disabled, 'Disabler in "if condition" must skip both blocks')
```

Comment:
Should it skip both blocks? I'd expect this code to print "missing header":

```meson
if cc.has_header('libavutil/hwcontext_drm.h', dependencies: libavutil):
    warning('has header')
else
    warning('missing header')
endif
```

Reply:
no, it skip both blocks. A disabler() is not a not-found dependency object.

Reply:
It is really a special type of object, unlike anything I've seen in other languages. You have understand how it works to use it, you could have surprises ;-)